server/sonar-server/src/main/java/org/sonar/server/projectanalysis/ws/SearchResults.java

```diff
@@ -111,14 +111,13 @@ String getProjectUuid() {
     private void filterByCategory() {
       ListMultimap<String, String> eventCategoriesByAnalysisUuid = events.stream()
         .collect(Collectors.index(EventDto::getAnalysisUuid, EventDto::getCategory));
-      AtomicInteger count = new AtomicInteger();
+      Predicate<SnapshotDto> byCategory = a -> eventCategoriesByAnalysisUuid.get(a.getUuid()).contains(request.getCategory().getLabel());
+      this.countAnalyses = (int) analyses.stream().filter(byCategory).count();
       this.analyses = analyses.stream()
-        .filter(a -> eventCategoriesByAnalysisUuid.get(a.getUuid()).contains(request.getCategory().getLabel()))
-        .peek(a -> count.incrementAndGet())
+        .filter(byCategory)
         .skip(Paging.offset(request.getPage(), request.getPageSize()))
```

Comment:
note that skip and limit and short-circuiting operations so peek was applied only on the subset of stream even if put before. That's where the bug comes from